design/ceph-mon-pv.md

```diff
@@ -0,0 +1,97 @@
+# Ceph monitor PV storage
+
+**Target version**: Rook 1.1
+
+## Overview
+
+Currently all of the storage for Ceph monitors (data, logs, etc..) is provided
+using HostPath volume mounts. Supporting PV storage for Ceph monitors in
+environments with dynamically provisioned volumes (AWS, GCE, etc...) will allow
+monitors to migrate without requiring the monitor state to be rebuilt, and
+avoids the operational complexity of dealing with HostPath storage.
+
+The general approach taken in this design document is to augment the CRD with a
+persistent volume claim template describing the storage requirements of
+monitors. This template is used by Rook to dynamically create a volume claim for
+each monitor.
+
+## Monitor storage specification
+
+The monitor specification in the CRD is updated to include a persistent volume
+claim template that is used to generate PVCs for monitor database storage.
+
+```go
+type MonSpec struct {
+	Count                int  `json:"count"`
+	PreferredCount       int  `json:"preferredCount"`
+	AllowMultiplePerNode bool `json:"allowMultiplePerNode"`
+	VolumeClaimTemplate  *v1.PersistentVolumeClaim
+}
+```
+
+The `VolumeClaimTemplate` is used by Rook to create PVCs for monitor storage.
+The current set of template fields used by Rook when creating PVCs are
+`StorageClassName` and `Resources`. Rook follows the standard convention of
+using the default storage class when one is not specified in a volume claim
+template. If the storage resource requirements are not specified in the claim
+template, then Rook will use a default value. This is possible because unlike
+the storage requirements of OSDs (xf: StorageClassDeviceSets), reasonable
```

Comment:
What does "(xf: StorageClassDeviceSets)" mean here?

Reply:
it's a cross reference to other new pvc work going on. in terms of good technical writing, this is definitely noise.